src/auth/reference_modules/oauth.py

```diff
@@ -0,0 +1,158 @@
+import time
+
+import jwt
+
+import requests
+
+
+def validate_jwt_token(token: str, scheme: str, client_id, tenant_id, type="access"):
+    jwks_uri = None
+    if scheme == "oauth-entra-id":
+        jwks_uri = f"https://login.microsoftonline.com/{tenant_id}/discovery/v2.0/keys"
+    else:
+        return {"valid": False, "error": "Unsupported scheme"}
+
+    jwks = requests.get(jwks_uri).json()
+
+    header = jwt.get_unverified_header(token)
+    if "alg" not in header or header["alg"] != "RS256":
+        return {"valid": False, "error": "Invalid algorithm in header"}
+
+    kid = header["kid"]
+
+    decoded_token = None
+    for jwk in jwks["keys"]:
+        if kid == jwk["kid"]:
+            try:
+                public_key = jwt.algorithms.RSAAlgorithm.from_jwk(json.dumps(jwk))
+                decoded_token = jwt.decode(token, key=public_key, algorithms=["RS256"], audience=client_id)
+            except Exception as e:
+                return {"valid": False, "error": e.args}
+
+    if decoded_token is None:
+        return {"valid": False, "error": "Matching kid not found"}
+
+    if "aud" not in decoded_token or decoded_token["aud"] != client_id:
+        return {"valid": False, "error": "Invalid audience"}
+
+    if decoded_token.get("exp", None) < int(time.time()):
+        return {"valid": False, "error": "Token expired"}
+
+    return {"valid": True, f"{type}_token": decoded_token}
+
+
+def _parse_response(response):
+    return dict(token.split("=") for token in response.split(";"))
+
+
+def _load_role_mappings(raw_role_mappings: str) -> dict:
+    if raw_role_mappings:
+        role_mapping = {}
+        raw_role_mappings = raw_role_mappings.split(";")
+        for mapping in raw_role_mappings:
+            idp_role, mg_role = mapping.split(":")
+            role_mapping[idp_role.strip()] = mg_role.strip()
+        return role_mapping
+    return {}
+
+
+def _load_config_from_env(scheme: str):
+    config = {}
+
+    if scheme == "oauth-entra-id":
+        config["id_issuer"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_ISSUER", "")
+        config["authorization_url"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_AUTHORIZATION_URL", "")
+        config["token_url"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_TOKEN_URL", "")
+        config["user_info_url"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_USER_INFO_URL", "")
+        config["client_id"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_CLIENT_ID", "")
+        config["client_secret"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_CLIENT_SECRET", "")
+        config["callback_url"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_CALLBACK_URL", "")
+        config["scope"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_SCOPE", "")
+        config["tenant_id"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_TENANT_ID", "")
+        config["role_mapping"] = _load_role_mappings(os.environ.get("AUTH_OAUTH_ENTRA_ID_ROLE_MAPPING", ""))
+
+    elif scheme == "oauth-okta":
+        config["id_issuer"] = os.environ.get("AUTH_OAUTH_OKTA_ISSUER", "")
```

Comment:
let's change all env vars to be prefixed with MEMGRAPH_

Reply:
Done in #2049. @imilinovic You may merge the changes from that PR’s branch :point_up: